src/parsers/perf/perfparser.cpp

```diff
@@ -1035,12 +1070,51 @@ class PerfParserPrivate : public QObject
             const auto attribute = attributes.value(event.type);
             if (attribute.type == static_cast<quint32>(AttributesDefinition::Type::Tracepoint)) {
                 Data::Tracepoint tracepoint;
-                tracepoint.time = event.time;
+                tracepoint.time = sample.time;
                 tracepoint.name = strings.value(attribute.name.id);
+
                 if (tracepoint.name != QLatin1String("sched:sched_switch")) {
                     // sched_switch events are handled separately already
                     tracepointResult.tracepoints.push_back(tracepoint);
                 }
+
+                if (exspectedTracepointName[sample.tid].isEmpty()) {
+                    addTracepointToTracepointStack(sample.tid, tracepoint);
+                } else {
+                    if (tracepoint.name == exspectedTracepointName[sample.tid].last()) {
+                        const auto startTracepoint = tracepointStack[sample.tid].back();
+
+                        bottomUpResult.addTracepointCost(tracepointCostIdLookup[startTracepoint.name], sample.frames,
+                                                         tracepoint.time - startTracepoint.time);
+
+                        tracepointStack[sample.tid].pop_back();
+                        exspectedTracepointName[sample.tid].pop_back();
+                    } else {
+                        // TODO: find solution for this
```

Comment:
as I mentioned, how did you get this?

but yes, e.g. in the face of lost events (you'll see an error/warning during perf record) this can happen. or if we only trace enter and not the exit events. or if we have user tracepoints and those don't get emitted in the correct order. 

so one way or another this needs to be handled gracefully. I'd just say emit a warning or such and override the previous start entry

Reply:
I used `perf record -o /home/lieven/perf.data --call-graph dwarf,8192 --event cycles:u,syscalls:sys_enter_openat,syscalls:sys_exit_openat --aio --sample-cpu /home/lieven/KDAB/hotspot/build/tests/test-clients/c-syscalls/c-syscalls`.
Hotspot reports lost events.

Reply:
right, that's somewhat expected then. try to limit the syscalls (e.g. only look at openat or smiilar). or enable compression via `-z` to resolve the lost chunks.

but what you are seeing here is a "reality" we need to live with in the parser frontend